lib/SILOptimizer/Analysis/AliasAnalysis.cpp

```diff
@@ -488,24 +488,24 @@ static bool typedAccessTBAAMayAlias(SILType LTy, SILType RTy,
   // Tuples do not alias non-tuples.
   bool LTyTT = LTy.is<TupleType>();
   bool RTyTT = RTy.is<TupleType>();
-  if ((LTyTT && !RTyTT) || (!LTyTT && RTyTT))
+  if (LTyTT != RTyTT)
     return false;
 
   // Structs do not alias non-structs.
   StructDecl *LTyStruct = LTy.getStructOrBoundGenericStruct();
   StructDecl *RTyStruct = RTy.getStructOrBoundGenericStruct();
-  if ((LTyStruct && !RTyStruct) || (!LTyStruct && RTyStruct))
+  if ((LTyStruct != NULL) != (RTyStruct != NULL))
```

Comment:
```suggestion
  if ((bool)LTyStruct != (bool)RTyStruct)
```

Reply:
Ditto the other analogous conditions.

Reply:
My personal opinion.
`LTyStruct != NULL` makes it clear `LTyStruct` is a pointer. This notation makes it easier to understand condition without need to be deep into context.
`(bool)LTyStruct` isn't that clear. Reader needs more context to understand the condition.
Suggested change doesn't make code clearer.
But for sure it's matter of personal preference and should be regulated by style guide.
@AnthonyLatsis are you sure I should apply change you suggested?

Reply:
I don’t recall any stylistic recommendations on this in the LLVM coding standards, static casts are just the most common pattern in the codebase (`!ptr` vs. `ptr != nullptr` is also a static cast, albeit implicit) for these things. In my opinion, it’s clear that these are pointers by looking at their declarations, which are right next to the `if` statement, but I’m not here to insist. Just please make sure you use `nullptr` rather than `NULL`.

Reply:
Agreed that an explicit nullptr check is warranted here. @valeriyvan please update this condition to use nullptr.

Reply:
done

Reply:
You may have forgotten to push.

Reply:
done